### PR TITLE
Empty collection request optimization

### PR DESCRIPTION
**Pull request by trueboroda**

### CODE REVIEW

File: VirtoCommerce.OrderModule.Data/Repositories/OrderRepositoryImpl.cs [hunk 1]
```diff
@@ -246,51 +247,77 @@ protected override void OnModelCreating(DbModelBuilder modelBuilder)
...
-            if (responseGroup.HasFlag(CustomerOrderResponseGroup.WithInPayments))
-            {
-                var inPayments = InPayments.Where(x => ids.Contains(x.CustomerOrderId)).ToArray();
-                var paymentsIds = inPayments.Select(x => x.Id).ToArray();
-                var paymentDiscounts = Discounts.Where(x => paymentsIds.Contains(x.PaymentInId)).ToArray();
-                var paymentTaxDetails = TaxDetails.Where(x => paymentsIds.Contains(x.PaymentInId)).ToArray();
-                var paymentAddresses = Addresses.Where(x => paymentsIds.Contains(x.PaymentInId)).ToArray();
-                var transactions = Transactions.Where(x => paymentsIds.Contains(x.PaymentInId)).ToArray();
-            }
+            CustomerOrderEntity[] result = Array.Empty<CustomerOrderEntity>();
```
Comment (artem-dudarev) [inline VirtoCommerce.OrderModule.Data/Repositories/OrderRepositoryImpl.cs]: var

Comment (trueboroda) [reply VirtoCommerce.OrderModule.Data/Repositories/OrderRepositoryImpl.cs]: yes)

File: VirtoCommerce.OrderModule.Data/Repositories/OrderRepositoryImpl.cs [hunk 2]
```diff
@@ -246,51 +247,78 @@ protected override void OnModelCreating(DbModelBuilder modelBuilder)
         public virtual CustomerOrderEntity[] GetCustomerOrdersByIds(string[] ids,
             CustomerOrderResponseGroup responseGroup)
         {
-            var result = CustomerOrders.Where(x => ids.Contains(x.Id)).ToArray();
-            var orderDiscounts = Discounts.Where(x => ids.Contains(x.CustomerOrderId)).ToArray();
-            var orderTaxDetails = TaxDetails.Where(x => ids.Contains(x.CustomerOrderId)).ToArray();
```
Comment (artem-dudarev) [inline VirtoCommerce.OrderModule.Data/Repositories/OrderRepositoryImpl.cs]: Remove

File: VirtoCommerce.OrderModule.Data/Repositories/OrderRepositoryImpl.cs [hunk 3]
```diff
@@ -246,51 +247,78 @@ protected override void OnModelCreating(DbModelBuilder modelBuilder)
...
+                        var paymentsIds = inPayments.Select(x => x.Id).ToArray();
+
+                        if (paymentsIds.Any())
+                        {
+                            var paymentDiscounts = Discounts.Where(x => paymentsIds.Contains(x.PaymentInId)).ToArray();
+                            var paymentTaxDetails = TaxDetails.Where(x => paymentsIds.Contains(x.PaymentInId)).ToArray();
+                            var paymentAddresses = Addresses.Where(x => paymentsIds.Contains(x.PaymentInId)).ToArray();
+                            var transactions = Transactions.Where(x => paymentsIds.Contains(x.PaymentInId)).ToArray();
+                        }
+
```
Comment (artem-dudarev) [inline VirtoCommerce.OrderModule.Data/Repositories/OrderRepositoryImpl.cs]: Remove

File: VirtoCommerce.OrderModule.Data/Repositories/OrderRepositoryImpl.cs [hunk 4]
```diff
@@ -246,51 +247,78 @@ protected override void OnModelCreating(DbModelBuilder modelBuilder)
...
+                        var lineItems = LineItems.Where(x => ids.Contains(x.CustomerOrderId))
+                            .OrderByDescending(x => x.CreatedDate).ToArray();
+                        var lineItemIds = lineItems.Select(x => x.Id).ToArray();
+
+                        if (lineItemIds.Any())
+                        {
+                            var lineItemDiscounts = Discounts.Where(x => lineItemIds.Contains(x.LineItemId)).ToArray();
+                            var lineItemTaxDetails = TaxDetails.Where(x => lineItemIds.Contains(x.LineItemId)).ToArray();
+                        }
+
```
Comment (artem-dudarev) [inline VirtoCommerce.OrderModule.Data/Repositories/OrderRepositoryImpl.cs]: Remove

File: VirtoCommerce.OrderModule.Data/Repositories/OrderRepositoryImpl.cs [hunk 5]
```diff
@@ -246,51 +247,78 @@ protected override void OnModelCreating(DbModelBuilder modelBuilder)
...
+                        if (shipmentIds.Any())
+                        {
+                            var shipmentDiscounts = Discounts.Where(x => shipmentIds.Contains(x.ShipmentId)).ToArray();
+                            var shipmentTaxDetails = TaxDetails.Where(x => shipmentIds.Contains(x.ShipmentId)).ToArray();
+                            var addresses = Addresses.Where(x => shipmentIds.Contains(x.ShipmentId)).ToArray();
+                            var shipmentItems = ShipmentItems.Where(x => shipmentIds.Contains(x.ShipmentId)).ToArray();
+                            var packages = ShipmentPackagesPackages.Include(x => x.Items)
+                                .Where(x => shipmentIds.Contains(x.ShipmentId)).ToArray();
+                        }
+
```
Comment (artem-dudarev) [inline VirtoCommerce.OrderModule.Data/Repositories/OrderRepositoryImpl.cs]: Remove

**Comment by yecli** [comment]
@trueboroda Please address review comments after the holidays.

**Pull request merged**